src/bandersnatch/mirror.py

```diff
@@ -493,6 +587,174 @@ def _load(self) -> None:
     def _save(self) -> None:
         self.statusfile.write_text(str(self.synced_serial), encoding="ascii")
 
+    def gen_data_requires_python(self, release: Dict) -> str:
+        if "requires_python" in release and release["requires_python"] is not None:
+            return f' data-requires-python="{html.escape(release["requires_python"])}"'
+        return ""
+
+    def generate_simple_page_for_package(self, package: Package) -> str:
```

Comment:
I don't get the advantage here? Can't you just not call this when using a Package() object?

Reply:
My thought process is that ```Package``` is/should be just a data structure with metadata about the package and releases, while the task of implementing the simple API (including the index) should be handled by a single dedicated class.

So, right now, the task of writing HTML is spit between package.py and mirror.py - mirror.py writes the [index page](https://github.com/pypa/bandersnatch/blob/master/src/bandersnatch/mirror.py#L373). My thought is that there should just be a ```SimpleMetadataWriter``` class or something which is responsible for both - it takes a list/queue/iterable of packages and turns it into a full repository, handling all the HTML writing both for the packages and for the index.

I'm definitely not married to any one particular idea though, there's a reason I ask before putting a ton of time into going in a particular direction :)